core/src/main/java/org/eclipse/hono/util/RegistrationConstants.java

```diff
@@ -158,17 +160,20 @@ public static JsonObject getRegistrationJson(final String action, final String t
     }
 
     public static JsonObject getRegistrationJson(final String action, final String tenantId, final String deviceId, final JsonObject payload) {
-        return getRegistrationJson(action, tenantId, deviceId, null, payload);
+        return getRegistrationJson(action, tenantId, deviceId, null, null, payload);
     }
 
-    public static JsonObject getRegistrationJson(final String action, final String tenantId, final String deviceId, final String key, final JsonObject payload) {
+    public static JsonObject getRegistrationJson(final String action, final String tenantId, final String deviceId, final String key, final String value, final JsonObject payload) {
         final JsonObject msg = new JsonObject();
         msg.put(APP_PROPERTY_ACTION, action);
         msg.put(APP_PROPERTY_DEVICE_ID, deviceId);
         msg.put(APP_PROPERTY_TENANT_ID, tenantId);
         if (key != null) {
             msg.put(APP_PROPERTY_KEY, key);
         }
+        if (value != null) {
```

Comment:
This sounds like we would accept a *find* query with a *key* and a *value* as usual but the code executing the search would then run into a `ClassCastException` in cases where the key's value is complex. Is that what you are proposing?